packages/back-end/src/controllers/stripe.ts

```diff
@@ -4,106 +4,142 @@ import {
   APP_ORIGIN,
   STRIPE_PRICE,
   STRIPE_WEBHOOK_SECRET,
-  STRIPE_DEFAULT_COUPON,
 } from "../util/secrets";
 import { Stripe } from "stripe";
 import { AuthRequest } from "../types/AuthRequest";
 import {
   updateOrganization,
-  updateOrganizationByStripeId,
+  findOrganizationByStripeCustomerId,
 } from "../models/OrganizationModel";
-import { createOrganization } from "../models/OrganizationModel";
 import { getOrgFromReq } from "../services/organizations";
+import { updateSubscription } from "../services/stripe";
 const stripe = new Stripe(STRIPE_SECRET || "", { apiVersion: "2020-08-27" });
 
-async function updateSubscription(subscription: string | Stripe.Subscription) {
-  // Make sure we have the full subscription object
-  if (typeof subscription === "string") {
-    subscription = await stripe.subscriptions.retrieve(subscription);
-  }
+let priceData: Stripe.Price;
```

Comment:
Caching priceData in memory as that doesn't (or shouldn't) really change.

I have opted to not cache the subscriptionData in memory, as I haven't figured out a reliable method to ensure the data isn't stale. I had to create a `stripe.ts` services file to handle updating the stripe subscription (so I could re-use the logic in `organizations.ts` when I handle adding/removing users & invites).

Currently, we're only hitting Stripe's  subscription API on /billing and /team (when the InviteModal is opened).

The 'BillingErrorBanner` component keys off of data in our database, as opposed to data from Stripe. (Not sure if that's best practice). But doing so ensures that we're not hitting the Stripe subscription API on every page load.

Reply:
@jdorn Let me know if the updated implementation is what you were thinking. I was under the impression that the memory would be unique to each visitor (like frontend memory) but realize that is not the case. Thank you - learned a new pattern here.